storage/src/mmr/mem.rs

```diff
@@ -90,139 +120,24 @@ impl<H: CHasher> Mmr<H> {
-                        end_tree_with_element = item;
-                        break;
-                    }
-                }
-            } else {
-                hashes.push(self.nodes[(item.0 - self.oldest_remembered_pos) as usize].clone());
-            }
-        }
-        assert!(start_tree_with_element.0 != u64::MAX);
-        assert!(end_tree_with_element.0 != u64::MAX);
-
-        // For the trees containing elements in the range, add left-sibling hashes of nodes along
-        // the leftmost path, and right-sibling hashes of nodes along the rightmost path, in
-        // decreasing order of the position of the parent node.
-        let left_path_iter = PathIterator::new(
-            start_element_pos,
-            start_tree_with_element.0,
-            start_tree_with_element.1,
-        );
-
-        let mut siblings = Vec::<(u64, u64)>::new();
-        if start_element_pos == end_element_pos {
-            // For the (common) case of a single element range, the right and left path are the
-            // same so no need to process each independently.
-            siblings.extend(left_path_iter);
-        } else {
-            let right_path_iter = PathIterator::new(
-                end_element_pos,
-                end_tree_with_element.0,
-                end_tree_with_element.1,
-            );
-            // filter the right path for right siblings only
-            siblings.extend(right_path_iter.filter(|(parent_pos, pos)| *parent_pos == *pos + 1));
-            // filter the left path for left siblings only
-            siblings.extend(left_path_iter.filter(|(parent_pos, pos)| *parent_pos != *pos + 1));
-
-            // If the range spans more than one tree, then the hashes must already be in the correct
-            // order. Otherwise, we enforce the desired order through sorting.
-            if start_tree_with_element.0 == end_tree_with_element.0 {
-                siblings.sort_by(|a, b| b.0.cmp(&a.0));
-            }
-        }
-        hashes.extend(
-            siblings
-                .iter()
-                .map(|(_, pos)| self.nodes[(*pos - self.oldest_remembered_pos) as usize].clone()),
-        );
-        Ok(Proof {
-            size: self.size(),
-            hashes,
-        })
-    }
-
     /// Returns the position of the oldest element that must be retained by this MMR in order to
     /// preserve its ability to generate proofs for new elements. This is the position of the
-    /// right-most leaf in the left-most peak. For the example tree in mod.rs, this would be node
-    /// 11.
+    /// tallest peak.
     pub fn oldest_required_element(&self) -> u64 {
         match self.peak_iterator().next() {
             None => {
                 // Degenerate case, only happens when MMR is empty.
                 0
             }
-            Some((pos, height)) => {
-                // Rightmost leaf position in a tree is the position of its peak minus its height.
-                pos - height as u64
```

Comment:
Was this a bug?

Reply:
not a bug technically in that nothing was broken, but it was just being more conservative than necessary.